src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace System.Security.Permissions
+{
+    public sealed partial class PrincipalPermission : System.Security.IPermission, System.Security.ISecurityEncodable, System.Security.Permissions.IUnrestrictedPermission
+    {
+        public PrincipalPermission(System.Security.Permissions.PermissionState state) { }
+        public PrincipalPermission(string name, string role) { }
+        public PrincipalPermission(string name, string role, bool isAuthenticated) { }
+        public System.Security.IPermission Copy() { return default(System.Security.IPermission); }
+        public void Demand() { }
```

Comment:
What API on ClaimsIdentity gives you back a ClaimsIdentity for the current OS user?

Regardless, why not WindowsIdentity?  It's available.  It works on Mono.  Code uses it.  What's the gigantic downside to making the basic cases "just work"?